src/sources/postgresql_metrics.rs

```diff
@@ -0,0 +1,851 @@
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_checkpoint_total",
+                counter!(row_get_value::<i64>(&row, "buffers_checkpoint")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_clean_total",
+                counter!(row_get_value::<i64>(&row, "buffers_clean")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_maxwritten_clean_total",
+                counter!(row_get_value::<i64>(&row, "maxwritten_clean")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_backend_total",
+                counter!(row_get_value::<i64>(&row, "buffers_backend")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_backend_fsync_total",
+                counter!(row_get_value::<i64>(&row, "buffers_backend_fsync")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_buffers_alloc_total",
+                counter!(row_get_value::<i64>(&row, "buffers_alloc")?),
+                tags!(self.tags),
+            ),
+            self.create_metric(
+                "pg_stat_bgwriter_stats_reset",
+                gauge!(row_get_value::<DateTime<Utc>>(&row, "stats_reset")?.timestamp()),
+                tags!(self.tags),
+            ),
+        ])
+    }
+
+    fn create_metric(
+        &self,
+        name: &str,
+        value: MetricValue,
+        tags: BTreeMap<String, String>,
+    ) -> Metric {
+        Metric {
+            name: name.into(),
+            namespace: self.namespace.clone(),
+            timestamp: Some(Utc::now()),
+            tags: Some(tags),
+            kind: MetricKind::Absolute,
+            value,
+        }
+    }
+}
+
+fn row_get_value<'a, T: FromSql<'a>>(row: &'a Row, key: &'static str) -> Result<T, CollectError> {
+    row.try_get::<&str, T>(key)
+        .map_err(|source| CollectError::PostgresGetValue { source, key })
+}
+
+fn config_to_endpoint(config: &Config) -> String {
+    let mut params: Vec<(&'static str, String)> = vec![];
+
+    // options
+    if let Some(options) = config.get_options() {
+        params.push(("options", options.to_string()));
+    }
+
+    // application_name
+    if let Some(name) = config.get_application_name() {
+        params.push(("application_name", name.to_string()));
+    }
+
+    // ssl_mode, ignore default value (SslMode::Prefer)
+    match config.get_ssl_mode() {
+        SslMode::Disable => params.push(("sslmode", "disable".to_string())),
+        SslMode::Prefer => {} // default, ignore
+        SslMode::Require => params.push(("sslmode", "require".to_string())),
+        _ => {} // non_exhaustive
```

Comment:
Maybe we could put the same warning here that we have for `get_target_session_attrs()`?

Reply:
Yes! Missed this one =(